File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_map = {}
        length = 0
        result = 0
        for r in range(len(s)):
            if s[r] in char_map:
                length = max(char_map[s[r]] + 1, length)
            char_map[s[r]] = r
            result = max(result, r - length + 1)
        return result
                

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0;
        int maxProfit = 0;
        while(l < prices.size())
        {
            int r = l + 1;
            while(r < prices.size())
            {
                maxProfit = max(maxProfit, prices[r] - prices[l]);
                r += 1;
            }
            l += 1;
        }
        return maxProfit;
    }
};
